core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -78,11 +78,11 @@ object ByteBufferMessageSet {
   }
 
   /** Deep iterator that decompresses the message sets and adjusts timestamp and offset if needed. */
-  def deepIterator(wrapperMessageAndOffset: MessageAndOffset): Iterator[MessageAndOffset] = {
+  def deepIterator(wrapperMessageAndOffset: MessageAndOffset): CloseableIterator[MessageAndOffset] = {
```

Comment:
For the old clients, our official public interface includes ByteBufferMessageSet, but it's the one in the kafka.javaapi.message package. So, we can change the interface here if needed.